### ISSUE TITLE
Editing storage file incorrectly causes all data to be lost

### ISSUE THREAD

**Issue by peiran18** [peiran18/pe#6]
After editing the json file to have two copies of data as such:

      "name": "John Doe",
      "phone": "98765432",
      "email": "johnd@example.com",
      "address": "311,Clementi Ave 2, #02-25",
      "tags": [
        "colleagues",
        "friends"
      ]

Upon opening the application again, all the data is gone.

**Comment by nus-se-script** [comment]
# Team's Response
Thank you for bringing this to our attention!

However, according to: https://github.com/nus-cs2103-AY2324S1/forum/issues/510 , 
> Recovering from improper edits to the data file (or other config files) was not an expectations at v1.4, unless you specifically stated such a feature in the UG.

This is also the exact way AB3 responds to this. Under functionality bugs: 
> Handling manual edits to the data file: AB3 UG specifies the current level of support for manually editing the data file. At least that level of support should be supported in the new product as well.

## The 'Original' Bug

[The team marked this bug as a duplicate of the following bug]

># Lack of warning when data storage file format is incorrect
>
>## Steps to reproduce
>1. Edit the data file to include duplicate people
>2. Launch application
>3. `exit` application
>
>## Description
>
>After step 3, the data file is empty, and previous data is permanently deleted. As it is a constraint for this application to allow editing of data file, should the application warn the user should the storage file be wrong, perhaps indicating in the example above that there are duplicate people? Without the lack of warning, many commands that result in the data file being saved can cause all previous data to be permanently edited should the user make a mistake in editing the file. Additionally, it might be possible to provide a backup of the incorrect data to give users the opportunity to fix it.
>
>## Reason for Severity
>
>This case might not be too rare, as users should be able to edit the data file. Furthermore, the consequences are quite detrimental, as it involves permanent data loss. Hence, `Medium` was chosen.
>
>
>
>
>
>
><hr>
><sub>[original: nus-cs2103-AY2324S1/pe-interim#1477] [original labels: type.FeatureFlaw severity.Medium]</sub>
>
>
>

## Their Response to the 'Original' Bug

[This is the team's response to the above 'original' bug]

>Thank you for bringing this to our attention!
>
>However, according to: https://github.com/nus-cs2103-AY2324S1/forum/issues/510 , 
>> Recovering from improper edits to the data file (or other config files) was not an expectations at v1.4, unless you specifically stated such a feature in the UG.
>
>This is also the exact way AB3 responds to this. Under functionality bugs: 
>> Handling manual edits to the data file: AB3 UG specifies the current level of support for manually editing the data file. At least that level of support should be supported in the new product as well.
>
>We also think that a warning message can be helpful. However, since our product does not crash on launch, we think that this is not in scope for now.
# Items for the Tester to Verify
## :question: Issue duplicate status

Team chose to mark this issue as a duplicate of another issue (as explained in the _**Team's response**_ above)

- [ ] I disagree

**Reason for disagreement:** [replace this with your explanation]

<catcher-end-of-segment><hr>
## :question: Issue response

Team chose [`response.NotInScope`]

- [ ] I disagree

**Reason for disagreement:** [replace this with your explanation]

<catcher-end-of-segment><hr>
## :question: Issue type

Team chose [`type.FeatureFlaw`]
Originally [`type.FunctionalityBug`]

- [ ] I disagree

**Reason for disagreement:** [replace this with your explanation]

<catcher-end-of-segment><hr>
## :question: Issue severity

Team chose [`severity.Medium`]
Originally [`severity.High`]

- [ ] I disagree

**Reason for disagreement:** [replace this with your explanation]

<catcher-end-of-segment><hr>
